From 39a62a1f336b2672ae7187bb56311e23d5f09306 Mon Sep 17 00:00:00 2001
From: Jason Hung <jhung@globalscaletechnologies.com>
Date: Tue, 3 Apr 2018 11:36:45 +0800
Subject: [PATCH 2/2] build: add compile option to build secondary boot image

add compile option 'SECONDARY_BOOTIMG' to build secondary boot image
, the default value is enabled.
---
 plat/marvell/a3700/common/a3700_common.mk | 10 +++++++++-
 plat/marvell/marvell.mk                   |  4 ++++
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/plat/marvell/a3700/common/a3700_common.mk b/plat/marvell/a3700/common/a3700_common.mk
index 0dc187e..cc45cb6 100644
--- a/plat/marvell/a3700/common/a3700_common.mk
+++ b/plat/marvell/a3700/common/a3700_common.mk
@@ -181,6 +181,7 @@ endif
 #	@mv -t $(BUILD_PLAT) $(TIM_IMAGE) $(DOIMAGE_CFG) $(TIMN_IMAGE) $(TIMNCFG) $(WTMI_IMG)
 #	@if [ "$(MARVELL_SECURE_BOOT)" = "1" ]; then mv -t $(BUILD_PLAT) $(WTMI_ENC_IMG) OtpHash.txt; fi
 #	@find . -name "*.txt" | grep -E "CSK[[:alnum:]]_KeyHash.txt|Tim_msg.txt|TIMHash.txt" | xargs rm -f
+ifeq ($(SECONDARY_BOOTIMG),1)
 	@echo
 	@echo "Building secondary flash image"
 	$(TIMBUILD) $(TIMBLD2ARGS)
@@ -205,15 +206,22 @@ endif
 	@if [ -e "$(TIMNCFG)" ]; then $(DOIMAGETOOL) -r $(TIMNCFG); fi
 	@if [ "$(MARVELL_SECURE_BOOT)" = "1" ]; then sed -i 's|$(WTMI_IMG)|$(WTMI_ENC_IMG)|1;s|$(BOOT_IMAGE)|$(BOOT_ENC_IMAGE)|1;' $(TIMNCFG); fi
 	$(TIM2IMG) $(TIM2IMGARGS) -o $(BUILD_PLAT)/$(FLASH_IMAGE_SEC)
+endif
 	@mv -t $(BUILD_PLAT) $(TIM_IMAGE) $(DOIMAGE_CFG) $(TIMN_IMAGE) $(TIMNCFG) $(WTMI_IMG) $(WTMI_SYSINIT_IMG) $(WTMI_MULTI_IMG)
 	@if [ "$(MARVELL_SECURE_BOOT)" = "1" ]; then mv -t $(BUILD_PLAT) $(WTMI_ENC_IMG) OtpHash.txt; fi
 	@find . -name "*.txt" | grep -E "CSK[[:alnum:]]_KeyHash.txt|Tim_msg.txt|TIMHash.txt" | xargs rm -f
 	@echo
 	@echo "Combine primary and secondary image"
 	@cp ${BUILD_PLAT}/${FLASH_IMAGE_PRI} ${BUILD_PLAT}/primary.img
-	@cp ${BUILD_PLAT}/${FLASH_IMAGE_SEC} ${BUILD_PLAT}/secondary.img
 	@truncate -s 2M ${BUILD_PLAT}/primary.img
+ifeq ($(SECONDARY_BOOTIMG),1)
+	@cp ${BUILD_PLAT}/${FLASH_IMAGE_SEC} ${BUILD_PLAT}/secondary.img
 	@truncate -s 2M ${BUILD_PLAT}/secondary.img
 	@cat ${BUILD_PLAT}/primary.img ${BUILD_PLAT}/secondary.img > ${BUILD_PLAT}/${FLASH_IMAGE}
 	@echo "Output combine image: ${BUILD_PLAT}/${FLASH_IMAGE}"
 	@rm ${BUILD_PLAT}/primary.img ${BUILD_PLAT}/secondary.img
+else
+	@cp ${BUILD_PLAT}/primary.img ${BUILD_PLAT}/${FLASH_IMAGE}
+	@echo "Output image: ${BUILD_PLAT}/${FLASH_IMAGE}"
+	@rm ${BUILD_PLAT}/primary.img
+endif
diff --git a/plat/marvell/marvell.mk b/plat/marvell/marvell.mk
index 6d289d6..88fe8c3 100644
--- a/plat/marvell/marvell.mk
+++ b/plat/marvell/marvell.mk
@@ -14,6 +14,10 @@ FLASH_IMAGE			:= flash-image.bin
 MARVELL_SECURE_BOOT		:= 0
 $(eval $(call add_define,MARVELL_SECURE_BOOT))
 
+# Build secondary bootloader image
+SECONDARY_BOOTIMG		:= 1
+$(eval $(call add_define,SECONDARY_BOOTIMG))
+
 # Enable compilation for Palladium emulation platform
 PALLADIUM			:= 0
 $(eval $(call add_define,PALLADIUM))
-- 
1.9.1

